builder/codebuild/builder.go

```diff
@@ -0,0 +1,76 @@
+package codebuild
+
+import (
+	"fmt"
+	"io"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/client"
+	"github.com/aws/aws-sdk-go/service/codebuild"
+	"github.com/remind101/conveyor/builder"
+	"golang.org/x/net/context"
+)
+
+// Builder is a builder.Builder implementation that runs the build in a docker
+// container.
+type Builder struct {
+	codebuild *codebuild.CodeBuild
+}
+
+// NewBuilder returns a new Builder backed by the docker client.
+func NewBuilder(config client.ConfigProvider) *Builder {
+	return &Builder{
+		codebuild: codebuild.New(config),
+	}
+}
+
+// NewBuilderFromEnv returns a new Builder with a docker client
+// configured from the standard Docker environment variables.
+func NewBuilderFromEnv() (*Builder, error) {
+	return &Builder{}, nil
+}
+
+// Build executes the docker image.
+func (b *Builder) Build(ctx context.Context, w io.Writer, opts builder.BuildOptions) (image string, err error) {
+
+	log.Printf("CODEBUILD BUILD STARTED")
+
+	params := &codebuild.StartBuildInput{
+		ProjectName:   aws.String("codebuild_test"),
```

Comment:
I'm assuming we'll probably be using a codebuild project per repo, so we'll need to:

1. Check if the project exists.
2. If not, create the project.

It would be good to make the project name configurable. A good way to do that in go is to use a [text/template](https://golang.org/pkg/text/template/), and pass it `opts` as the data:

```golang
t := template.Must(template.New("project").Parse("conveyor-{{.Repository}}"))
b := new(bytes.Buffer)
err := t.Execute(b, opts)
```